src/Device/Driver/LX_EOS/LXEosDeclare.cpp

```diff
@@ -0,0 +1,218 @@
+  memset(data.gps, 0x00, sizeof(data.gps));
+
+  // Number of turnpoints (excluding start and finish)
+  data.num_of_tp = declaration.Size() - 2;
+
+  // Takeoff point (set to all zeroes, as it is not useful)
+  data.prg[0] = 3;
+  data.lat[0] = 0;
+  data.lon[0] = 0;
+  CopyStringSpacePadded(data.name[0], _T("TAKEOFF"), 9);
+
+  for (uint8_t i = 1; i <= declaration.Size(); i++) {
+    data.prg[i] = 1;
+    data.lat[i] = ConvertCoord(declaration.GetLocation(i - 1).latitude);
+    data.lon[i] = ConvertCoord(declaration.GetLocation(i - 1).longitude);
+    CopyStringSpacePadded(data.name[i], declaration.GetName(i - 1), 9);
+  }
+
+  /*
+   * Setting the landing seems to not work.
+   * Eos will either show name of waypoint that used to be on that position in
+   * list before declaration, or "NO_NAME". It is more likely a bug in the Eos
+   * firmware than in the structure of the command. It seems to not cause any
+   * practical issues, so let's ignore it.
+   */
+  data.prg[declaration.Size() + 1] = 2;
+  data.lat[declaration.Size() + 1] = 0;
+  data.lon[declaration.Size() + 1] = 0;
+  CopyStringSpacePadded(data.name[declaration.Size() + 1], _T("LANDING"), 9);
+
+  // Set the remaining entries to zeroes
+  for (uint8_t i = declaration.Size() + 2; i < 12; i++) {
+    data.prg[i] = 0;
+    data.lat[i] = 0;
+    data.lon[i] = 0;
+    memset(data.name[i], 0x00, 9);
+  }
+
+  // CRC is calculated only from data bytes (excluding the SYN and CMD)
+  // Using poly 0x69 and initial 0xFF
+  data.crc = Calculate8bitCRC(
+    reinterpret_cast<uint8_t*>(&data) + 2, sizeof(data) - 3, 0xFF, 0x69);
+  std::span<std::byte> message(reinterpret_cast<std::byte*>(&data),
+                               sizeof(data));
+
+  return WriteAndWaitForACK(message, env);
+}
+
+bool
+LXEosDevice::SendObsZone(uint8_t tp_nr,
+                         const Declaration& declaration,
+                         OperationEnvironment& env)
+{
+  if (tp_nr < 1 || tp_nr > declaration.Size())
+    return false;
+
+  EosObsZoneStruct data;
+  const auto& turnpoint = declaration.turnpoints.at(tp_nr - 1);
+
+  data.uiTpNr = tp_nr;
+  data.uiR1 = ToLE32(turnpoint.radius);
+
+  if (tp_nr == 1)                       // Start
+    data.uiDirection = 2;               // Next
+  else if (tp_nr == declaration.Size()) // Finish
+    data.uiDirection = 3;               // Previous
+  else                                  // Turnpoint
+    data.uiDirection = 0;               // Symmetric
+
+  data.bIsLine = (turnpoint.shape == Declaration::TurnPoint::LINE) ? 1 : 0;
+
+  /* We cannot be sure about the format of float on the platform, hardcoded
+   * bytes are used for the angles */
+
+  // Bytes of float values for 45, 90 and 180 degree angles in radians
+  uint8_t angle180[] = { 0xdb, 0x0f, 0x49, 0x40 };
+  uint8_t angle90[] = { 0xdb, 0x0f, 0xc9, 0x3f };
+  uint8_t angle45[] = { 0xdb, 0x0f, 0x49, 0x3f };
+
+  // Init A21 and A2 to zero
+  memset(&data.fA21, 0, 4);
+  memset(&data.fA2, 0, 4);
```

Comment:
4 is the size of float in the structure that is transmitted to the vario (if it wasn't, the static_assert of structure size would fail). 
In the previous review, you pointed out that the number format may not be guaranteed on all architectures, so I wanted to explicitly set all bytes to zero rather than expecting that float zero is represented like that on any platform.

Reply:
This is all true, but it has nothing to do with my complaint. In this line of code, you don't see what `fA21` is, and you don't know how big it is. This obscures the code. You could have used `sizeof(data.fa21)` or just do `data.fA21 = {};`, but you didn't - instead you hard-coded `4` which makes your code rather obscure.